app_check/src/FirebaseAppCheck.cs

```diff
@@ -79,8 +87,18 @@ public sealed class FirebaseAppCheck {
   ///
   /// This method should be called before initializing the Firebase App.
   public static void SetAppCheckProviderFactory(IAppCheckProviderFactory factory) {
+    if (appCheckFactory == factory) return;
+
     appCheckFactory = factory;
-    // TODO(amaurice): Clear the provider map when the factory changes?
+    // Clear out the Providers that were previously made
+    providerMap.Clear();
+
+    // Register the callback for C++ SDK to use that will reach this factory.
```

Comment:
I wonder a bit about why we need to conditionally set and clear getTokenDelegate when we set and clear the C# factory.

Reply:
It isn't necessarily something we have to do, but just doing it for convenience of reducing the back and forth of C++ / Unity.  By setting it to null, the C++ part of the implementation can know that the C# side was cleared out, and pass errors back without having to reach up to C# first, which also involves putting things on the update queue.